src/bsoncxx/CMakeLists.txt

```diff
@@ -117,85 +117,72 @@ if (BSONCXX_POLY_USE_BOOST)
     find_package(Boost 1.56.0 REQUIRED)
 endif()
 
-# We define two libraries: the normal library and the testing-only library.  The testing-only
-# library does not get installed, but the tests link against it instead of the normal library.  The
-# only difference between the two libraries is that BSONCXX_TESTING is defined in the testing-only
-# library (this define enables special testing-only behavior).
-add_library(bsoncxx
-    ${bsoncxx_sources}
-)
-add_library(bsoncxx_testing
-    ${bsoncxx_sources}
-)
-target_compile_definitions(bsoncxx_testing PUBLIC BSONCXX_TESTING)
-
-set_target_properties(bsoncxx_testing PROPERTIES
-    OUTPUT_NAME bsoncxx-testing
-)
+function(bsoncxx_add_library TARGET OUTPUT_NAME LINK_TYPE)
```

Comment:
Since nearly the same target is created two or three times, I wrapped the creation in a CMake function.

Reply:
Nice!

Reply:
This function, along with `bsoncxx_install` should be factored out into a separate file, perhaps `cmake/BsoncxxUtil.cmake`, which can then be included with `include(BsoncxxUtil)` near the top of this `CMakeLists.txt`.

Reply:
@rcsanchez97, sure, happy to do so.